### PR TITLE
Overdue: Make sure to chunk before updating in database

### LINKED ISSUE #147: IcngaDB fails with: Prepared statement contains too many placeholders - Overdue indicators not chunked
## Description

In high loaded environment i'm facing IcngaDB crash with the following errors:

```
Mar 31 08:56:44 p3i212t4-idb systemd: Started Icinga DB.
Mar 31 08:56:44 p3i212t4-idb icingadb: time="2020-03-31T08:56:44-07:00" level=info msg="Connecting to Redis"
Mar 31 08:56:44 p3i212t4-idb icingadb: time="2020-03-31T08:56:44-07:00" level=info msg="Connecting to MySQL"
Mar 31 08:56:44 p3i212t4-idb icingadb: time="2020-03-31T08:56:44-07:00" level=info msg="Redis connection established"
Mar 31 08:56:44 p3i212t4-idb icingadb: time="2020-03-31T08:56:44-07:00" level=info msg="Starting heartbeat listener"
Mar 31 08:56:44 p3i212t4-idb icingadb: time="2020-03-31T08:56:44-07:00" level=info msg="SQL connection established"
Mar 31 08:56:44 p3i212t4-idb icingadb: time="2020-03-31T08:56:44-07:00" level=info msg="Got initial environment." UUID=e727d600-79fe-4cfb-b636-83e5a7ccd7a8 context=HA environment=da39a3ee5e6b4b0d3255bfef95601890afd80709
Mar 31 08:56:44 p3i212t4-idb icingadb: time="2020-03-31T08:56:44-07:00" level=info msg="Taking over." UUID=e727d600-79fe-4cfb-b636-83e5a7ccd7a8 context=HA environment=da39a3ee5e6b4b0d3255bfef95601890afd80709
Mar 31 08:56:46 p3i212t4-idb icingadb: time="2020-03-31T08:56:46-07:00" level=fatal msg="Error 1390: Prepared statement contains too many placeholders"
Mar 31 08:56:46 p3i212t4-idb systemd: icingadb.service: main process exited, code=exited, status=1/FAILURE
Mar 31 08:56:46 p3i212t4-idb systemd: Unit icingadb.service entered failed state.
Mar 31 08:56:46 p3i212t4-idb systemd: icingadb.service failed.
```

## How to reproduce:
Currently i have lack of info for deeper diagnostics and not able to reproduce it, error occurs randomly on high loaded environment with 100k services per minute.

## Environment:

Standalone icinga2 master: 8 cores / 16 GB ram
2 Satellite zones in HA(4 icinga2 instances as satellites): 8 cores / 16 GB ram
1 IcingaDB + Redis on dedicated host: 8 cores / 16 GB ram
MySQL cluster on dedicated hosts: 8 cores / 16 GB ram
CentOS Linux release 7.7.1908 (Core) on each server
Amount of active services: 102010 services per minute

Please let me know if i can provide any additional debug information for you.
With best regards,
Dmitriy.

### PR DESCRIPTION
**Pull request by N-o-X**
fixes #147

### CODE REVIEW

File: configobject/statesync/overdue.go [hunk 1]
```diff
@@ -156,32 +156,43 @@ func updateOverdue(super *supervisor.Supervisor, objectType string, counter *uin
 
 // updateOverdueInDb sets objectType_state#is_overdue for ids to overdue.
 func updateOverdueInDb(super *supervisor.Supervisor, objectType string, observer prometheus.Observer, ids []interface{}, overdue bool) {
-	placeholders := make([]string, 0, len(ids))
-	for len(placeholders) < cap(placeholders) {
-		placeholders = append(placeholders, "?")
-	}
+	for _, c := range utils.ChunkInterfaces(ids, 1000) {
+		go func(chunk []interface{}) {
```
Comment (Al2Klimov) [inline configobject/statesync/overdue.go:160]: You're not waiting for this to finish. Are you sure that's a good idea?

File: configobject/statesync/overdue.go [hunk 2]
```diff
@@ -156,32 +156,43 @@ func updateOverdue(super *supervisor.Supervisor, objectType string, counter *uin
...
-		),
-		args...,
-	)
-	if errSE != nil {
-		super.ChErr <- errSE
-		return
+			log.WithFields(log.Fields{
+				"amount": len(chunk),
+				"type": objectType,
+				"overdue": overdue,
```
Comment (Al2Klimov) [inline configobject/statesync/overdue.go:180]: Formatting

File: configobject/statesync/overdue.go (hunk 1 above)
Comment (N-o-X) [reply configobject/statesync/overdue.go:160]: I'm not sure why we should wait here. Any benefits?

Comment (N-o-X) [reply configobject/statesync/overdue.go:160]: Okay, this is a reason:
https://github.com/Icinga/icingadb/blob/b20cb2e4ed4282335bd529952144a8241d24e5a1/configobject/statesync/overdue.go#L150

Anything else?

Comment (Al2Klimov) [reply configobject/statesync/overdue.go:160]: No, I've already requested all changes I'd like to see.

**Pull request merged**